aws/resource_aws_iot_role_alias_test.go

```diff
@@ -0,0 +1,332 @@
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccAWSIotRoleAlias_basic(t *testing.T) {
+	alias := acctest.RandomWithPrefix("RoleAlias-")
+	alias2 := acctest.RandomWithPrefix("RoleAlias2-")
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckAWSIotRoleAliasDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSIotRoleAliasConfig(alias),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSIotRoleAliasExists("aws_iot_role_alias.ra"),
+					resource.TestCheckResourceAttr(
+						"aws_iot_role_alias.ra", "credential_duration", "3600"),
+				),
+			},
+			{
+				Config: testAccAWSIotRoleAliasConfigUpdate1(alias, alias2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSIotRoleAliasExists("aws_iot_role_alias.ra"),
+					testAccCheckAWSIotRoleAliasExists("aws_iot_role_alias.ra2"),
+					resource.TestCheckResourceAttr(
+						"aws_iot_role_alias.ra", "credential_duration", "1800"),
+				),
+			},
+			{
+				Config: testAccAWSIotRoleAliasConfigUpdate2(alias2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSIotRoleAliasExists("aws_iot_role_alias.ra2"),
+				),
+			},
+			{
+				Config: testAccAWSIotRoleAliasConfigUpdate3(alias2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSIotRoleAliasExists("aws_iot_role_alias.ra2"),
+				),
+				ExpectError: regexp.MustCompile("Role alias .+? already exists for this account"),
+			},
+			{
+				Config: testAccAWSIotRoleAliasConfigUpdate4(alias2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSIotRoleAliasExists("aws_iot_role_alias.ra2"),
+				),
+			},
+			{
+				Config: testAccAWSIotRoleAliasConfigUpdate5(alias2),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSIotRoleAliasExists("aws_iot_role_alias.ra2"),
+					resource.TestMatchResourceAttr(
+						"aws_iot_role_alias.ra2", "role_arn", regexp.MustCompile(".+?bogus")),
+				),
+			},
+		},
+	})
+
+}
+
+func listIotRoleAliasPages(conn *iot.IoT, input *iot.ListRoleAliasesInput,
+	fn func(out *iot.ListRoleAliasesOutput, lastPage bool) bool) error {
+	for {
+		page, err := conn.ListRoleAliases(input)
+		if err != nil {
+			return err
+		}
+		lastPage := page.NextMarker == nil
+
+		shouldContinue := fn(page, lastPage)
+		if !shouldContinue || lastPage {
+			break
+		}
+		input.Marker = page.NextMarker
+	}
+	return nil
+}
+
+func testAccCheckAWSIotRoleAliasDestroy(s *terraform.State) error {
+	conn := testAccProvider.Meta().(*AWSClient).iotconn
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "aws_iot_role_alias" {
+			continue
+		}
+
+		alias := rs.Primary.Attributes["alias"]
+
+		input := &iot.ListRoleAliasesInput{
+			PageSize: aws.Int64(250),
+		}
+
+		var roleAlias string
+		err := listIotRoleAliasPages(conn, input, func(out *iot.ListRoleAliasesOutput, lastPage bool) bool {
```

Comment:
It seems like we should be able to use `getIotRoleAliasDescription()` here and `continue` on the correct error instead of implementing this custom function. e.g.

```go
_, err := getIotRoleAliasDescription(conn, rs.Primary.ID)

if isAWSErr(err, iot.ErrCodeResourceNotFoundException, "") {
  continue
}

return fmt.Errorf("IoT Role Alias (%s) still exists", rs.Primary.ID)
```